sktime/transformations/series/compose.py

```diff
@@ -96,3 +97,143 @@ def inverse_transform(self, Z, X=None):
         if not self.passthrough:
             z = self.transformer_.inverse_transform(z, X=None)
         return z
+
+
+class ColumnComposition(_SeriesToSeriesTransformer):
+    """
+    Parameters
+    ----------
+    transformer : series-to-series transformer to be applied to each series
+    columns : list of str
+            Names of columns that are supposed to be transformed
+    """
+
+    _required_parameters = ["transformer"]
+
+    def __init__(self, transformer, columns="all"):
+        self.transformer = transformer
+        self.columns = columns
+        self.transformers_ = None
+        super(ColumnComposition, self).__init__()
+
+    def fit(self, Z, X=None):
+        """
+        Iterates over columns (series) and applies the fit function of the transformer
+        """
+        z = check_series(Z, allow_numpy=False)
+        self._is_fitted = False
+
+        # univariate case
+        if isinstance(z, pd.Series):
+            self.transformer.fit(z, X)
+            self._is_fitted = True
+            return self
+        # multivariate case
+        else:
+            if self.columns == "all":
+                self.columns = z.columns
+
+            # make sure z contains all columns that the user wants to transform
+            Z_wanted_keys = set(self.columns)
+            Z_new_keys = set(z.columns)
+            difference = Z_wanted_keys.difference(Z_new_keys)
+            if len(difference) != 0:
+                raise ValueError("Missing columns" + str(difference) + "in Z.")
+
+            self.transformers_ = {}
+            for colname in self.columns:
+                transformer = clone(self.transformer)
+                self.transformers_[colname] = transformer
+                self.transformers_[colname].fit(z[colname], X)
+                # self.transformers_[colname].is_fitted = True
+            self._is_fitted = True
+            return self
+
+    def transform(self, Z, X=None):
+        """
+        calls transform on every single transformer (one transformer per series)
+        """
```

Comment:
I think we need also a proper docstring here, see e.g.  `HampelFilter`:
https://github.com/alan-turing-institute/sktime/blob/859ac7206ad8aad8ce418b391c521064083e30c3/sktime/transformations/series/outlier_detection.py#L63-L74